tests/core/filtering/test_contract_data_filters.py

```diff
@@ -0,0 +1,198 @@
+    else:
+        event_filter = create_filter(Emitter, [
+            'LogDynamicArgs', {
+                'filter': {
+                    'arg1': vals['matching']}}])
+
+    txn_hashes = []
+    txn_hashes.append(
+        emitter.functions.logDynamicArgs(
+            arg0=vals['matching'], arg1=vals['matching']).transact())
+    txn_hashes.append(
+        emitter.functions.logDynamicArgs(
+            arg0=vals['non_matching'][0], arg1=vals['non_matching'][0]).transact())
+    txn_hashes.append(
+        emitter.functions.logDynamicArgs(
+            arg0=vals['non_matching'][1], arg1=vals['non_matching'][1]).transact())
+    txn_hashes.append(
+        emitter.functions.logDynamicArgs(
+            arg0=vals['non_matching'][2], arg1=vals['non_matching'][2]).transact())
+    txn_hashes.append(
+        emitter.functions.logDynamicArgs(
+            arg0=vals['non_matching'][3], arg1=vals['non_matching'][3]).transact())
+
+    for txn_hash in txn_hashes:
+        wait_for_transaction(web3, txn_hash)
+
+    log_entries = event_filter.get_new_entries()
+    assert len(log_entries) == 1
+    assert log_entries[0]['transactionHash'] == txn_hashes[0]
+
+
+@pytest.mark.parametrize('call_as_instance', (True, False))
+@given(vals=fixed_values())
+def test_data_filters_with_fixed_arguments(
+        web3,
+        emitter,
+        Emitter,
+        wait_for_transaction,
+        emitter_event_ids,
+        call_as_instance,
+        create_filter,
+        vals):
+
+    if call_as_instance:
+        event_filter = create_filter(emitter, [
+            'LogQuadrupleArg', {
+                'filter': {
+                    'arg0': vals['matching'][0],
+                    'arg1': vals['matching'][1],
+                    'arg2': vals['matching'][2],
+                    'arg3': vals['matching'][3]}}])
+    else:
+        event_filter = create_filter(Emitter, [
+            'LogQuadrupleArg', {
+                'filter': {
+                    'arg0': vals['matching'][0],
+                    'arg1': vals['matching'][1],
+                    'arg2': vals['matching'][2],
+                    'arg3': vals['matching'][3]}}])
+
+    txn_hashes = []
+    txn_hashes.append(emitter.functions.logQuadruple(
+        which=5,
+        arg0=vals['matching'][0],
+        arg1=vals['matching'][1],
+        arg2=vals['matching'][2],
+        arg3=vals['matching'][3]).transact())
+    txn_hashes.append(emitter.functions.logQuadruple(
+        which=5,
+        arg0=vals['non_matching'][0],
+        arg1=vals['non_matching'][1],
+        arg2=vals['non_matching'][2],
+        arg3=vals['non_matching'][3]).transact())
+
+    for txn_hash in txn_hashes:
+        wait_for_transaction(web3, txn_hash)
+
+    log_entries = event_filter.get_new_entries()
+    assert len(log_entries) == 1
+    assert log_entries[0]['transactionHash'] == txn_hashes[0]
+
+
+@pytest.mark.xfail(reason="The regex based data filters do not work with dynamic sized types")
```

Comment:
What kind of failure do we get? Do we just not get the events we were looking for? It seems like we should explicitly raise an exception if we're incapable of generating the filter accurately. Then the test could be a `pytest.raises()` on that exception.